app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposerState.kt

```diff
@@ -20,403 +20,151 @@
-                        )
-                    )
-                }
-            }
-        }
-    }
+/**
+ * Class holding the whole UI state for the MessageComposer, this is the class that is used by the out-side world to give the control
+ * of the state to the parent Composables
+ */
+class MessageComposerStateHolder(
+    val messageComposerViewState: MutableState<MessageComposerViewState>,
+    val messageCompositionInputStateHolder: MessageCompositionInputStateHolder,
+    val messageCompositionHolder: MessageCompositionHolder,
+    val additionalOptionStateHolder: AdditionalOptionStateHolder,
+    val modalBottomSheetState: WireModalSheetState,
+    val tempWritableImageUri: Uri?,
+    val tempWritableVideoUri: Uri?
+) {
+    val messageComposition = messageCompositionHolder.messageComposition
 
-    fun showAttachmentOptions() = changeAttachmentOptionsVisibility(true)
-    fun hideAttachmentOptions() = changeAttachmentOptionsVisibility(false)
-    private fun changeAttachmentOptionsVisibility(newValue: Boolean) {
-        (messageComposeInputState as? MessageComposeInputState.Active)?.let { activeState ->
-            when (val currentType = activeState.type) {
-                is MessageComposeInputType.NewMessage -> {
-                    messageComposeInputState = activeState.copy(
-                        type = currentType.copy(
-                            attachmentOptionsDisplayed = newValue,
-                            richTextFormattingOptionsDisplayed = false
-                        )
-                    )
-                }
+    val isTransitionToKeyboardOnGoing
+        @Composable get() = additionalOptionStateHolder.additionalOptionsSubMenuState == AdditionalOptionSubMenuState.Hidden &&
+                !KeyboardHelper.isKeyboardVisible() &&
+                messageCompositionInputStateHolder.inputFocused
+    val additionalOptionSubMenuVisible
+        @Composable get() = additionalOptionStateHolder.additionalOptionsSubMenuState != AdditionalOptionSubMenuState.Hidden &&
+                !KeyboardHelper.isKeyboardVisible()
 
-                is MessageComposeInputType.SelfDeletingMessage -> {
-                    messageComposeInputState = activeState.copy(
-                        type = currentType.copy(
-                            attachmentOptionsDisplayed = newValue,
-                            richTextFormattingOptionsDisplayed = false
-                        )
-                    )
-                }
+    private var isKeyboardVisible = false
 
-                else -> {}
-            }
-        }
-    }
+    val isSelfDeletingSettingEnabled = messageComposerViewState.value.selfDeletionTimer !is SelfDeletionTimer.Disabled &&
+            messageComposerViewState.value.selfDeletionTimer !is SelfDeletionTimer.Enforced
 
-    fun messageComposeInputFocusChange(isFocused: Boolean) {
-        messageComposeInputState = messageComposeInputState.copyCurrent(inputFocused = isFocused)
+    fun toInActive() {
```

Comment:
```suggestion
    fun toInactive() {
```

Inactive is a single word.